primitive-types/impls/num-traits/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 Parity Technologies
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! num-traits support for uint.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[doc(hidden)]
+pub use num_traits;
+
+use core::fmt;
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum FromStrRadixErrKind {
+	Parse,
+	UnsupportedRadix,
+}
+
+#[derive(Debug)]
+enum FromStrRadixErrSrc {
+	Hex(uint::FromHexError),
+	Dec(uint::FromDecStrErr),
+}
+
+impl fmt::Display for FromStrRadixErrSrc {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		match self {
+			FromStrRadixErrSrc::Dec(d) => write!(f, "{}", d),
+			FromStrRadixErrSrc::Hex(h) => write!(f, "{}", h),
+		}
+	}
+}
+
+#[derive(Debug)]
+pub struct FromStrRadixErr {
```

Comment:
thank you for this clean design with opaque error type,
our policy is to have all new public types/methods/API documented (`///`)